src/NServiceBus.Core/Transports/Msmq/MsmqMessageDispatcher.cs

```diff
@@ -77,6 +77,8 @@ void ExecuteTransportOperation(TransportTransaction transaction, UnicastTranspor
 
                         var label = GetLabel(message);
 
+                        Task.Yield(); // MessageQueue.Send is a blocking operation
```

Comment:
You have to await a `Task.Yield`, so this has introduced a code path that is never awaited, which is going to cause all sorts of problems.